presto-common/src/main/java/com/facebook/presto/common/block/BlockUtil.java

```diff
@@ -326,4 +326,29 @@ public static int getNum128Integers(int length)
         }
         return num128Integers;
     }
+
+    /**
+     * Returns null if all blocks are already loaded, otherwise returns a new blocks array with all blocks loaded
+     */
```

Comment:
Can we return `blocks` if all blocks are already loaded? This will be consistent with our other existing lazy block loading interfaces.

Reply:
I suppose we could, but the usage pattern from the row based call sites are looking to inspect the return value to answer the question: "should I create a new `(Single)RowBlock`?" which seems more intuitive to me to have `null` stand-in as the "nothing changed" result instead of returning `blocks`. It also helps that returning `null` means that callers can't accidentally unconditionally wrap the result in a new instance (since null would blow up with an exception) and accidentally introduce a performance bug.

What are the other examples of lazy block loading interfaces you're referring to? Maybe if there's a strong enough existing idiom then returning `blocks` makes sense, but I'm not familiar with it the other examples you're referring to to decide.

Reply:
The most common one would be `Block.getLoadedBlock`. The implementations of this interface has the contract of returning the same block if it's already fully loaded. `RowBlock.getLoadedBlock` is a good example. Can we follow the same pattern?

Reply:
Sure, done.